src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesList = [];
+  const currentState = { ...state };
+
+  for (const action of actions) {
+    const actionType = action.type;
+
+    switch (actionType) {
+      case 'addProperties':
+        const actionData = action.extraData;
```

Comment:
you can also [destructure ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment?retiredLocale=pl)`extraData`